.github/workflows/ci.yml

```diff
@@ -1,7 +1,104 @@
+########################################################################################################
+# Build configurations
+# ====================
+#
+# full
+# ----
+# Description: Basic build with all components (TLX, C++, Cython). Has to be tested on every fully 
+#              supported platform. Close to user-experience, except that warnings are also treated 
+#              as errors during build and TLX is provided as externally build library (normal 
+#              installation builds both NetworKit and TLX in one step).
+# Script:      scripts/full.sh
+# Build-type:  Release
+#
+# 
+# full (variant)
+# ---------------------
+# Description: These builds are variants of full with different options.
+# Script:      scripts/full.sh
+# Variants:    native = Build for native architecture
+# 
+#         
+# core
+# ----
+# Description: Builds the C++ code into one shared library. Has also sanity checks enabled.
+# Script:      scripts/core.sh
+# Build-type:  Debug
+#
+# 
+# core (variant)
+# ---------------------
+# Description: These builds are variants of core with different options. 
+# Script:      scripts/core.sh
+# Variants:    release = Release-build
+#              non-monolithic = Each C++ module is a library
+#              C++17 = Check for C++17 conformance
+#              C++20 = Check for C++20 conformance
+# 
+#
+# clang-tidy
+# ----------
+# Description: Uses clang-tidy for several static analysis checks. These involve bug prone programming
+#              errors, possible modernizations and performance optiomizations.
+# Script:      scripts/clang_tidy.sh
+# Config:      /.clang-tidy
+# Build-type:  Debug
+#
+# 
+# coverage
+# --------
+# Description: Enables code-coverage, leak-sanitizers and tests in order to create cpp-coveralls data.
```

Comment:
Maybe specify address and leak sanitizers (address sanitizers are enabled by default if one sets `-DNETWORKIT_WITH_SANITIZERS=leak`).